[BUG] mvn test failed compiling rapids-4-spark-tests-next-spark_2.12

**Describe the bug**
Related to https://github.com/NVIDIA/spark-rapids/pull/3663

To reproduce `mvn test -Dbuildver=[any version >= 310+]` 

```
[2021-09-25T18:21:35.311Z] [INFO] --- scala-maven-plugin:4.3.0:testCompile (scala-test-compile-first) @ rapids-4-spark-tests-next-spark_2.12 ---

[2021-09-25T18:21:35.311Z] [INFO] Using incremental compilation using Mixed compile order

[2021-09-25T18:21:35.311Z] [INFO] Compiler bridge file: /var/lib/jenkins/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.12-1.3.1-bin_2.12.8__52.0-1.3.1_20191012T045515.jar

[2021-09-25T18:21:35.311Z] [INFO] Compiling 2 Scala sources to /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/target/test-classes ...

[2021-09-25T18:21:36.240Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/InsertPartition311Suite.scala:21: not found: type SparkQueryCompareTestSuite

[2021-09-25T18:21:36.497Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/InsertPartition311Suite.scala:26: not found: value SparkSessionHolder

[2021-09-25T18:21:36.754Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/InsertPartition311Suite.scala:30: not found: value testSparkResultsAreEqual

[2021-09-25T18:21:36.754Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/InsertPartition311Suite.scala:31: not found: value testName

[2021-09-25T18:21:36.754Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:38: not found: type SparkQueryCompareTestSuite

[2021-09-25T18:21:36.754Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:40: not found: value test

[2021-09-25T18:21:36.754Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:41: not found: value withCpuSparkSession

[2021-09-25T18:21:37.011Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:48: not found: value test

[2021-09-25T18:21:37.011Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:49: not found: value withCpuSparkSession

[2021-09-25T18:21:37.011Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:62: not found: value test

[2021-09-25T18:21:37.011Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:71: not found: value test

[2021-09-25T18:21:37.575Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:135: not found: value fail

[2021-09-25T18:21:37.575Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:144: not found: value withResource

[2021-09-25T18:21:37.575Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:174: not found: value withCpuSparkSession

[2021-09-25T18:21:37.575Z] [ERROR] [Error] /raid1/jenkins/workspace/rapids_gds-nightly/plugin/tests-spark310+/src/test/scala/com/nvidia/spark/rapids/Spark310ParquetWriterSuite.scala:184: not found: value fail

[2021-09-25T18:21:37.575Z] [ERROR] 15 errors found

[2021-09-25T18:21:37.575Z] [INFO] ------------------------------------------------------------------------

[2021-09-25T18:21:37.575Z] [INFO] Reactor Summary for RAPIDS Accelerator for Apache Spark Root Project 21.10.0-SNAPSHOT:

[2021-09-25T18:21:37.575Z] [INFO] 

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Root Project ... SUCCESS [  0.808 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin ..... SUCCESS [01:06 min]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin . SUCCESS [  8.289 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Shims SUCCESS [  0.897 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark SQL Plugin Spark 3.1.3 Shim SUCCESS [  4.669 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin SUCCESS [ 19.646 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Aggregator ..... SUCCESS [  0.063 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Distribution ... SUCCESS [  0.115 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark UDF Examples ... SUCCESS [  5.404 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Tests .......... SUCCESS [ 27.304 s]

[2021-09-25T18:21:37.575Z] [INFO] rapids-4-spark-integration-tests_2.12 .............. SUCCESS [ 14.134 s]

[2021-09-25T18:21:37.575Z] [INFO] rapids-4-spark-api-validation ...................... SUCCESS [  5.929 s]

[2021-09-25T18:21:37.575Z] [INFO] RAPIDS Accelerator for Apache Spark Tests For 3.1.X+ FAILURE [  2.841 s]

[2021-09-25T18:21:37.575Z] [INFO] ------------------------------------------------------------------------

[2021-09-25T18:21:37.575Z] [INFO] BUILD FAILURE

```

--- a/tests-spark310+/pom.xml
+++ b/tests-spark310+/pom.xml
@@ -63,6 +63,7 @@
             <groupId>com.nvidia</groupId>
             <artifactId>rapids-4-spark-tests_${scala.binary.version}</artifactId>
             <version>${project.version}</version>
+            <type>test-jar</type>
             <classifier>${spark.version.classifier}tests</classifier>
             <scope>test</scope>
         </dependency>

